docs/madr/decisions/083-meshservice-meshidentity-identity.md

```diff
@@ -0,0 +1,298 @@
+# Migration from one SpiffeID to another without a downtime using MeshService
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/14388
+
+## Context and Problem Statement
+
+MeshService stores identities in the object:
+
+```yaml
+spec:
+  identities:
+  - type: ServiceTag
+    value: test-server_default_kuma-demo_svc_80
+  - type: SpiffeID
+    value: spiffe://default.zone.mesh.local/ns/kuma-demo/sa/default
+```
+`ServiceTag` identity is a value that can be precomputed and set on a `MeshService` without checking whether mTLS is enabled.
+In the case of `MeshIdentity`, we set the `SpiffeID` based on the matching resource. We determine which `MeshIdentity` targets a specific `MeshService`, and based on that, we assign the corresponding identity.
+
+Later, this information is used to create the Envoy cluster configuration. Once a `MeshIdentity` is enabled and targets specific dataplanes, Kuma begins delivering identities to each sidecar. At this point, the `MeshService` is recalculated, new identities are added, and these are then translated into SAN matching.
+
+### Where is the issue?
+
+Let’s walk through the flow:
+
+Prerequisites:
+* `MeshService` is in `Exclusive` mode
+* There is constant traffic from the client to the service
+
+1. Mesh with legacy mTLS (`Mesh.mTLS`) is enabled
+2. Client and Server details
+* Client presents identity: `client_default_kuma-demo_svc_80`
+* Server presents identity: `server_default_kuma-demo_svc_80`
+
+MeshService of Client
```

Comment:
Does MeshService of Client play any role in this communication? I don't think it's relevant if `Client` also acts as server for someone else